quickwit-cli/src/lib.rs

```diff
@@ -391,6 +405,53 @@ fn display_statistics(
     Ok(())
 }
 
+/// ThroughputCalculator is used to calculate throughput on time interval.
+struct ThroughputCalculator {
+    /// clock for tracking the time
+    clock: Instant,
+    /// How long it takes before calculating the throughput.
+    window_size: u64,
+    /// Stores the previous value of processed bytes.
+    processed_bytes_from_last_window: usize,
+    /// Stores the caluclated throughput.
+    throughput: f64,
+}
+
+impl ThroughputCalculator {
+    /// Creates new instance.
+    pub fn new(clock: Instant, window_size: u64) -> Self {
+        Self {
+            clock,
+            window_size,
+            processed_bytes_from_last_window: 0,
+            throughput: 0f64,
+        }
+    }
+
+    /// Calculates the throughput.
+    pub fn calculate(&mut self, processed_bytes: usize) -> f64 {
+        let elapsed_secs = self.clock.elapsed().as_secs();
+
+        // Only recompute if we have reached the end of a window
+        // and the number of processed byte has changed.
+        if elapsed_secs % self.window_size == 0
+            && self.processed_bytes_from_last_window != processed_bytes
+        {
+            let window_processed_bytes = processed_bytes - self.processed_bytes_from_last_window;
+            self.throughput =
+                window_processed_bytes as f64 / 1_000_000f64 / self.window_size as f64;
+            self.processed_bytes_from_last_window = processed_bytes;
```

Comment:
There is a complicated big here.

If for one reason or another elapsed_secs skips 8s. For instance... It was 7s30 and the next tick it is 9s due to some lag in the terminal output.
Another possibility is we changed the refresh interval in the terminal.

What will happen?
Or maybe 
What happens then?